tests/test_fx/test_ckpt_solvers/test_ckpt_solver_perf.py

```diff
@@ -0,0 +1,108 @@
+import os
+from typing import Tuple
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision.models as tm
+from torch.fx import symbolic_trace
+import torch.fx
+from torch.optim import Adam
+from torch.nn import CrossEntropyLoss
+import colossalai
+from colossalai.fx.passes.meta_info_prop import MetaInfoProp
+from colossalai.fx import ColoGraphModule, ColoTracer
+from colossalai.fx.passes.algorithms import solver_rotor
+from colossalai.utils import free_port
+import pytest
+import time
+import numpy as np
+
+try:
+    from colossalai.fx.codegen import ActivationCheckpointCodeGen
+except:
+    from colossalai.fx.codegen import python_code_with_activation_checkpoint
+
+
+def _data_gen(batch_size: int, shape: Tuple[int, int, int], device='cuda'):
+    data = torch.rand(batch_size, *shape, device=device)
+    label = torch.empty(batch_size, dtype=torch.long, device=device).random_(1000)
+    return data, label
+
+
+def _test_forward(gm: torch.fx.GraphModule, bs: int = 32, num_steps: int = 5):
+    gm.train()
+    gm.cuda()
+    criterion = CrossEntropyLoss()
+    step_time = float('inf')
+    for _ in range(num_steps):
+        data, label = _data_gen(bs, (3, 224, 224))
+        torch.cuda.synchronize(device="cuda")
+        time0 = time.time()
+        output = gm(data)
+        loss = criterion(output, label)
+        loss.backward()
+        torch.cuda.synchronize(device="cuda")
+        time1 = time.time()
+        step_time = min(step_time, time1 - time0)
+        for child in gm.children():
+            for param in child.parameters():
+                param.grad = None
+        del data, label
+
+    gm.to("cpu")
+
+    return torch.cuda.max_memory_allocated(device="cuda") / 1024**2, step_time * 1.0e3
+
+
+@pytest.mark.skip(reason="no needed in pytest")
+def test_meta_info_prop(bs: int = 32, rank=0):
+    colossalai.launch(config={}, rank=rank, world_size=1, host='localhost', port=free_port(), backend='nccl')
+
+    for M in [
+            tm.resnet18, tm.resnet34, tm.resnet50, tm.resnet101, tm.resnet152, tm.densenet121, tm.densenet161,
+            tm.densenet169, tm.densenet201
+    ]:
+
+        torch.cuda.reset_peak_memory_stats()
+        print(f"Testing {M.__name__}")
```

Comment:
 remove logging and printing in unittests.
 Unitest is only used to verify the functionality.

Reply:
This one is a performance test of this unit, I simply skip it in pytest, maybe there is a directory for us to place those scripts?

Reply:
I will remove this script and put it in examples later when the solver is stable